### ISSUE TITLE
Not linting under gfortran 5.1

### ISSUE THREAD

**Issue by heidenfahrt** [AtomLinter/linter-gfortran#10]
While other linters are working perfectly, linter-gfortran does not linter anything in a simple *.f90 file. The gfortran version installed is 5.1.0. Might this be due to the new linter version?

**Comment by Arcanemagus** [comment]
The specs are passing under gfortran v6.2.0 and v5.4.1. It's certainly possible though that between v5.1.0 and v5.4.1 the output changed. Can you try running the following on your file in question to see what the output looks like?
```shell
gfortran -Wall -Wextra -fsyntax-only -J /tmp -I . ./foo.f90
```

**Comment by heidenfahrt** [comment]
This is the output I get:
```./TestFor.f90:3:20:
 
 PROGRAM TESTFORTRAN    ! Program head (every FORTRAN program starts with a program head)
                    1
Warning: Nonconforming tab character at (1) [-Wtabs]
./TestFor.f90:6:1:
 
  PRINT*, 'Hello World'
 1

Warning: Nonconforming tab character at (1) [-Wtabs]
./TestFor.f90:9:24:
 
 END PROGRAM TESTFORTRAN   ! End of program
                        1
Warning: Nonconforming tab character at (1) [-Wtabs] 
```

Where the code is simply:

````

! ..............................................................
PROGRAM TESTFORTRAN				! Program head (every FORTRAN program starts with a program head)
! ..............................................................

	PRINT*, 'Hello World'

! ..............................................................
END PROGRAM TESTFORTRAN			! End of program
! ..............................................................

**Comment by Arcanemagus** [comment]
Hmmm that should be working: https://regex101.com/r/JvC4bZ/1

Are you getting any error messages in the Console?

**Comment by heidenfahrt** [comment]
I do, they look very similar:
```
Object {type: "Warning", text: "Nonconforming tab character at (1) [-Wtabs]", filePath: "\Users\username\AppData\Local\Temp\atom-linter_4844qTOqoXSCSWZ9\TestFor.f90", range: Array[2]}
```
When I open it:

```
filePath
:
"\Users\feholub\AppData\Local\Temp\atom-linter_4844qTOqoXSCSWZ9\TestFor.f90"
key:"$LINTER:gfortran$LOCATION:\Users\feholub\AppData\Local\Temp\atom-linter_4844qTOqoXSCSWZ9\TestFor.f90$2$19$2$19$TEXT:Nonconforming tab character at (1) [-Wtabs]$HTML:$SEVERITY:warning$TYPE:Warning$CLASS:"
linterName:"gfortran"
range:Range
severity:"warning"
text:"Nonconforming tab character at (1) [-Wtabs]"
type:"Warning"
version:1
__proto__:Object
```

There are three of these messages which look very similar.
Can you explain me what to do with the regex?

**Comment by Luthaf** [comment]
>  Can you explain me what to do with the regex?

The code is using [this regex](https://github.com/AtomLinter/linter-gfortran/blob/21c0258dbd2355d1f3ecac73750fdf6458470058/lib/main.js#L22) to [match](https://github.com/AtomLinter/linter-gfortran/blob/21c0258dbd2355d1f3ecac73750fdf6458470058/lib/main.js#L52) gfortran output.

Could you share the version of atom-linter and linter-gfortran you are using?

**Comment by heidenfahrt** [comment]
Thank you for the explanation.
My linter version is 2.1.2, linter-gfortran is 0.4.0

**Comment by Luthaf** [comment]
OK. I can not reproduce the issue on linux.

I'll try to start a Windows VM when I can, to see what I can do. How did you install gfortran ?

**Comment by Arcanemagus** [comment]
That path looks like macOS, not Windows @Luthaf 😉.

**Comment by heidenfahrt** [comment]
I was using this installation: http://tdm-gcc.tdragon.net/download
I will now install the most recent version from here: http://mingw-w64.org/doku.php

**Comment by heidenfahrt** [comment]
I'm now running gfortran 6.3.0 using http://mingw-w64.org/doku.php. Compiling works fine but the errors reported remain the same.

**Comment by jimmyn** [comment]
I have the same problem.

```
>gfortran -Wall -Wextra -fsyntax-only -I . D:\work\Test.for
D:\work\Test.for:2:9:

       END PROGAM
         1
Error: Expecting END PROGRAM statement at (1)
f951.exe: Error: Unexpected end of file in 'D:\work\Test.for'
```

In console everything works fine but I dont see any errors in atom

**Comment by Luthaf** [comment]
The issue could come from [here](https://github.com/AtomLinter/linter-gfortran/blob/0090eae95b3a90f70e3718a52ec7f5b142cf7503/lib/main.js#L54), if for some reason the file path and the temporary file path are not the same on Windows (maybe because of the drive letter?).

What do you see in the console if you add `console.log(error.filePath); console.log(tmpFilename);` before the if?

**Comment by jimmyn** [comment]
@Luthaf yes, it seems that the problem is in drive latter:
```
\Users\Varvara\AppData\Local\Temp\atom-linter_43400rWCULigPnfT\Test.for
C:\Users\Varvara\AppData\Local\Temp\atom-linter_43400rWCULigPnfT\Test.for
```

**Comment by VEZY** [comment]
Hello,
unless I missed something during installation, I don't have any lintering in Atom either. 
I tried to execute gfortran -Wall -Wextra -fsyntax-only -J /tmp -I . foo.f90 on the same sample file as @heidenfahrt did and I have this output from gfortran (from the console):

`gfortran -Wall -Wextra -fsyntax-only -J /tmp -I . foo.f90
f951.exe: Warning: Nonexistent include directory '/tmp' [-Wmissing-include-dirs]
foo.f90:3:20:

 PROGRAM TESTFORTRAN    ! Program head (every FORTRAN program starts with a program head)
                    1
Warning: Nonconforming tab character at (1) [-Wtabs]
foo.f90:6:1:

  PRINT*, 'Hello World'
 1
Warning: Nonconforming tab character at (1) [-Wtabs]
foo.f90:9:24:

 END PROGRAM TESTFORTRAN   ! End of program
                        1
Warning: Nonconforming tab character at (1) [-Wtabs]`

But there is nothing showing in Atom (zero Linter Errors, warnings or Infos). Here is my version of gfortran:
GNU Fortran (x86_64-posix-seh-rev2, Built by MinGW-W64 project) 6.3.0

**Comment by Luthaf** [comment]
@heidenfahrt, @jimmyn or @VEZY: could one of you check-out and try #15 locally, to see if this fix the issue?

**Comment by Luthaf** [comment]
This should be fixed by the 0.5 version. Feel free to re-open this issue or to create a new one if something does not work.
